test/e2e/upgrade_test.go

```diff
@@ -41,6 +43,18 @@ func upgradeTestSuite(t *testing.T) {
 	require.NoError(t, err, "error testing Windows workloads")
 	defer testCtx.deleteJob(testerJob.Name)
 
+	// check BYOH enabled
+	if gc.numberOfBYOHNodes >= 1 {
+		// ensure the presence of a running workload in BYOH node
+		byohDeployment, err := testCtx.deployByohWorkload()
+		require.NoError(t, err, "error deploying BYOH workload")
+		// cleanup workload
+		defer func(testCtx *testContext, name string) {
+			err := testCtx.deleteDeployment(name)
+			require.NoErrorf(t, err, "error deleting deployment %s", name)
```

Comment:
since deleteDeployment returns an error you can just check on the function and defer that
require.NoError(t, err, testCtx.deleteDeployment(name))

Reply:
Nice!

Reply:
We should not fail the test on cleanup errors

Reply:
Sounds good, will adjust.